File: Microservices/ForumsService/Models/View/ForumSearchView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace ForumServices.Models.View
{
    public class ForumSearchView
    {
        [JsonPropertyName("totalItem")]
        public int TotalItem { get; set; }
        [JsonPropertyName("totalPage")]
        public int TotalPage { get; set; }
        [JsonPropertyName("currentPage")]
        public int CurrentPage { get; set; }
        [JsonPropertyName("nbItemPerPage")]
        public int NbItemPerPage { get; set; }
        [JsonPropertyName("nameFilter")]
        public string NameFilter { get; set; }
        [JsonPropertyName("sescFilter")]
        public string DescFilter { get; set; }
        [JsonPropertyName("forumSearch")]
        public List<ForumView> ForumSearch { get; set; }
        [JsonPropertyName("totalItemCurrent")]
        public int TotalItemCurrent { get; internal set; }
    }
}

File: UI/forum-app/src/models/forum/ForumSearchView.ts
import ForumView from "./ForumView";

export default class ForumSearchView {
  public totalItemCurent:number;
  public totalItem: number;
  public totalPage: number;
  public currentPage: number;
  public nbItemPerPage: number;
  public nameFilter: string;
  public descFilter: string;
  public forumSearch: Array<ForumView>;
}
